### DISCUSSION TITLE
No registered user after reboot

### DISCUSSION THREAD

**Discussion by adevoss** [inspircd/inspircd#2019] [Support]
Still installing and testing Inspircd.
Finally installed Anope after modifying source code.

Confirmed registration with the code from the mail.
Nick was registered.

After reboot my nick is not registered anymore.
(Debian 11 - inspircd_3.15.0.deb11u2_amd64.deb from GitHub)

Where does inspircd store data? The /var/lib/inspircd is empty.
Can all data be stored in a mysql database?
I added the mysql module to the config and created a database.

**Comment by sadiepowell** [comment]
Account registration is handled by Anope which is a separate program to InspIRCd and does not use any InspIRCd modules or data directories. Typically Anope writes its data to a flat file database but I'm not sure where that would be stored with your setup. Try asking [here](https://forum.anope.org/index.php?board=17.0) or irc.anope.org #anope.

**Reply by adevoss** [reply]
Anope is installed and uses a mysql database. The registered nick is stored in the database.
How does Inspircd gets this information to check if a nick exists/is registered and if the password is correct?
Is there a way to see if Anope does not give the information or if Inspircd does not retreive the information?

**Reply by sadiepowell** [reply]
InspIRCd doesn't retrieve anything from the services database. When you log into an account InspIRCd just passes your login details to Anope over the server link and then if your login succeeded Anope sends back metadata with your account name.

You can look up whether Anope knows about your account with `/msg NickServ INFO <nick>`. If its in the database but Anope can't find it with that command then make sure you're not using Anope 2.0.8 as that version contained a regression where it would corrupt your MySQL database. If that is the case then you'll need to restore from a backup.

**Comment by adevoss** [comment]
Using 2.0.9-1 from Debian so that should be fine.

The INFO command first returns "isn't registered"
Right after registration, information about nick is returned with different date (current) than stored in db..
After reboot "isn't registered" is returned again.
These two pieces of information both are in the anope log so I have to debug anope and check which sql statements are executed in mysql. maybe an mysql connection problem
Hopefully that will lead to a solution.

For now I'm only testing so no need to restore.

Thanks so far.

**Comment by adevoss** [comment]
It was a mysql connection problem (replaced localhost with 127.0.0.1)
